dataactbroker/fileRoutes.py

```diff
@@ -102,3 +102,17 @@ def get_protected_files():
         """ Return signed URLs for all help page files """
         fileManager = FileHandler(request, isLocal=IS_LOCAL, serverPath=SERVER_PATH)
         return RouteUtils.run_instance_function(fileManager, fileManager.getProtectedFiles)
+
+    @app.route("/v1/generate_file/", methods=["POST"])
+    @permissions_check
+    def generate_file():
+        """ Generate file from external API """
+        fileManager = FileHandler(request, isLocal=IS_LOCAL, serverPath=SERVER_PATH)
+        return RouteUtils.run_instance_function(fileManager, fileManager.generateFile)
+
+    @app.route("/v1/check_generation_status/", methods=["POST"])
+    @permissions_check
```

Comment:
Maybe out of scope, but I would assume that a "check" function would respond to a `GET` rather than a `POST`. I'd expect something like:

http://example.com/v1/generation_status/D1?session=mytoken